docs: create NavLink cookbook example

Adbib opened the pull request:
The NavLink component enhances navigation by getting isPending and isActive status to styling your links

# Overview

<!--
The Qwik Team and Qwik Community are grateful for all PRs that improve Qwik. Thank you for your time and effort! Please be aware that not all PRs can be merged, but PRs that meet the following criteria will receive the highest priority:

a) Fixes to the core, and

b) Framework functionality that can only be achieved by the core.

If your functionality can be delivered as a 3rd-Party Community Add-On, we encourage that route as it will likely provide a faster path to adoption.

If you feel your functionality is of high value to everybody in the Qwik Community, we encourage socializing it in the Qwik Discord channels as the core team may take this up for inclusion in the core.

_— Build primitives is our mantra_

-->

# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [x] Docs / tests / types / typos

# Description
When building navigation in Qwik, you may want to know when a link is active or pending navigation. The standard <a> and <Link> components don't provide this.

The NavLink component enhances navigation links by adding:
- **Active Status** - Apply a class when the link matches the current location. 
- **Pending Status** - Apply a class during navigation when linking to the target location.

# Use cases and why

<!-- Actual / expected behavior if it's a bug -->

- 1. navbar
- 2. sidebar

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

gioboa reviewed (changes requested):
You need to add the new section here.
`packages/docs/src/routes/docs/menu.md`
`packages/docs/src/routes/docs/cookbook/index.mdx`
Can you update this files too pls?

gioboa commented on packages/docs/src/routes/docs/cookbook/nav-link/index.mdx:
````diff
@@ -0,0 +1,67 @@
+
+This allows NavLink to know the active and pending state automatically based on navigation.
+
+```tsx
````
Can you change this snippet with a component you can check the other examples with this imports
`import CodeSandbox, {CodeFile} from '../../../../components/code-sandbox/index.tsx';`
This helps us ensure alignment over time.

Adbib replied:
sorry i don't understand

gioboa replied:
The other cookbook examples are using `CodeSandbox` and `CodeFile` to ensure alignment over time.
Can you do the same pls?

Adbib replied:
done

gioboa commented:
> You need to add the new section here. `packages/docs/src/routes/docs/menu.md` `packages/docs/src/routes/docs/cookbook/index.mdx` Can you update this files too pls?

Can you do the change in this PR and not in a different one pls?

gioboa commented:
@Adbib I fixed up the docs. I added my suggestions too.
Are we considering all the cases for isActive?
what about href="/about/" 
is it considered?

Adbib commented:
> @Adbib I fixed up the docs. I added my suggestions too. Are we considering all the cases for `isActive`prop? what about href="/about/" is it considered?

yes is it

gioboa commented:
I added the CodeSandbox example and it doesn't seem to work and I can't understand how verify isPending 🤔 
<img width="1360" alt="Screenshot 2023-12-26 at 17 26 54" src="https://github.com/BuilderIO/qwik/assets/35845425/5a0b077c-7720-4127-b3fa-e90bfa708816">

Adbib commented:
> I added the CodeSandbox example and it doesn't seem to work and I can't understand how verify isPending 🤔 <img alt="Screenshot 2023-12-26 at 17 26 54" width="1360" src="https://private-user-images.githubusercontent.com/35845425/292894567-5a0b077c-7720-4127-b3fa-e90bfa708816.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2MTk0MjcsIm5iZiI6MTcwMzYxOTEyNywicGF0aCI6Ii8zNTg0NTQyNS8yOTI4OTQ1NjctNWEwYjA3N2MtNzcyMC00MTI3LWIzZmEtZTkwYmZhNzA4ODE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI2VDE5MzIwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwYmI1ZDk5YWYxZmVlZjY2MzI0MmY1Y2ZjMmFkZWQ4N2Y5YjFmNTQ2MzJmNjc1Mjk2YzAyZmVhYWI4MTNmMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g-hS-CrRUsJPAf2LuD98X3kK_Jw7h9mYoPK-41de09k">

I think because we do a full-page refresh with a tag.
if you try it with < Link > that does client-side navigation it should work

gioboa commented:
Can you verify the missing active and pending classes pls?
If your implementation is with Link you can change it

Adbib commented:
> Can you verify the missing active and pending classes pls? If your implementation is with Link you can change it

its work now

gioboa commented:
I can't verify it now, can you send me a screenshot pls?

Adbib commented:
> I can't verify it now, can you send me a screenshot pls?

this video with implementation of the same code : 
https://github.com/BuilderIO/qwik/assets/17273701/61f42dec-578e-4174-8cea-00af5e2e2d06

gioboa commented:
@Adbib I cleaned up the example. I removed the pending state b/c I wan't able to make it works.
If for you it's ok I will merge it.

Adbib commented:
> @Adbib I cleaned up the example. I removed the pending state b/c I wasn't able to make it works. If for you it's ok I will merge it.

okay lets merge it for now untiil we find a way to pendding
